test/asami/query_internals_test.cljc

```diff
@@ -363,4 +363,12 @@
       (is (vartest? p))
       (is (vartest? v)))))
 
+(deftest test-column-renaming
+  (is (= '{:cols [?x ?z ?y]}
+         (meta (q/query-entry '{:find [?x :data :as ?z ?y]
```

Comment:
Since this is an integration test, it would be appropriate to see that the projection worked correctly as well.
i.e. The result of the query should be `([1 :data 2])`